tensorflow/compiler/xla/client/client_library.h

```diff
@@ -47,7 +48,7 @@ class LocalClientOptions {
   LocalClientOptions(se::Platform* platform = nullptr,
                      int number_of_replicas = 1,
                      int intra_op_parallelism_threads = -1,
-                     std::set<int> device_set = {-1});
+                     const absl::optional<std::set<int>>& allowed_devices = {});
```

Comment:
Nit, prefer `nullopt` ("explicit is better than implicit", and `{}` looks like we're initializing the set to an empty set, which is the opposite of what we're doing.)